src/main/java/com/epam/task/figures/entities/Pyramid.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.task.figures.entities;
+
+import com.epam.task.figures.generators.IdGenerator;
+
+import java.util.Objects;
+
+public class Pyramid extends Point{
+    private Point a;
+    private Point b;
+    private Point c;
+    private Point d;
+    private Point e;
+    private final int id;
+    IdGenerator generator =new IdGenerator();
```

Comment:
alway write modificator such as private
IdGenerator should be here (single responsibility)
pass genereted id in constructor